src/storage-preview/azext_storage_preview/_params.py

```diff
@@ -333,6 +339,115 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
                         ' be uploaded.')
         c.ignore('destination')
 
+    for item in ['stats', 'exists', 'metadata show', 'metadata update']:
+        with self.argument_context('storage queue {}'.format(item)) as c:
+            c.extra('timeout', help='Request timeout in seconds. Applies to each call to the service.', type=int)
+
+    for item in ['exists', 'generate-sas', 'create', 'delete', 'metadata show', 'metadata update']:
+        with self.argument_context('storage queue {}'.format(item)) as c:
+            c.extra('q_name', queue_name_type, options_list=('--name', '-n'), required=True)
+
+    with self.argument_context('storage queue create') as c:
+        c.argument('fail_on_exist', help='Specify whether to throw an exception if the queue already exists.')
+
+    with self.argument_context('storage queue delete') as c:
+        c.argument('fail_not_exist', help='Specify whether to throw an exception if the queue doesn\'t exist.')
+
+    with self.argument_context('storage queue generate-sas') as c:
+        from .completers import get_storage_acl_name_completion_list
+        t_queue_permissions = self.get_sdk('_models#QueueSasPermissions', resource_type=CUSTOM_DATA_STORAGE_QUEUE)
+
+        c.argument('ip', type=ipv4_range_type,
+                   help='Specify the IP address or range of IP addresses from which to accept requests. '
+                        'Support only IPv4 style addresses.')
+        c.argument('expiry', type=get_datetime_type(True),
+                   help='Specify the UTC datetime (Y-m-d\'T\'H:M\'Z\') at which the SAS becomes invalid. Do not '
+                        'use if a stored access policy is referenced with --policy-name that specifies this value.')
+        c.argument('start', type=get_datetime_type(True),
+                   help='Specify the UTC datetime (Y-m-d\'T\'H:M\'Z\') at which the SAS becomes valid. Do not use '
+                        'if a stored access policy is referenced with --policy-name that specifies this value. '
+                        'Default to the time of the request.')
```

Comment:
consider grouping these arguments as track 1 https://github.com/Azure/azure-cli/blob/e836dd2365f1b81b73784088f9a60ca49a8a3b02/src/azure-cli/azure/cli/command_modules/storage/_params.py#L1353.

These arguments could be utilized among all storage services.

Reply:
done